### DISCUSSION TITLE
using a different version of ruby

### DISCUSSION THREAD

**Discussion by tswetonic** [community/community#27166] [Actions]
<p>I’m working on an action to run CI for our rails app. I was using actions/setup-ruby@v1 successfully, however our app won’t run using a ruby higher than 2.3.4. I’m trying to install ruby 2.3.x and running into various issues. Anyone have a recommendation on how to install ruby 2.3.4? I’m running my job on ubuntu-latest. </p>

**Comment by DannyBen** [comment]
<p>I know this is probably not what you want to hear, but perhaps it is time to deal with the real issue at hand? 2.3 is EOL and things are only going to get harder for you the longer you wait.</p>
<p>Perhaps updating the app is a better way to invest your time, and eliminate technical debt rather than adding to it?</p>
<p>That said, it this is not an option, then perhaps you can simply follow the usual Ruby installation steps - maybe with RVM?</p>

**Comment by tswetonic** [comment]
<p>Indeed, upgrading ruby is important but not happening at the moment.</p>

**Comment by tswetonic** [comment]
<p>btw I’m using rbenv to install ruby. The Gemfile is bundled fine, but when I try to run a rake task, gems aren’t found. I’ve tried various things, like setting GEM_PATH, setting BUNDLE_PATH, etc. Haven’t found the solution yet. This is more of a ruby question at this point, but I was hoping somebody has run into this.</p>

**Comment by DannyBen** [comment]
<p>I tried installing ruby 2.3 inside the Actions runner, using snap, just to see if I can provide you with a solution. The installation succeeded, but running ruby --version still shows newer version.</p>
<p>Perhaps you can play with it further.</p>
<pre><code>name: Test
on: [push]

jobs:
  main:
    name: Rush demo
    runs-on: ubuntu-latest

    steps:
    - name: Install Ruby
      run: |
        sudo snap install ruby --classic --channel=2.3/stable
        sudo snap refresh
        ruby --version
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/tswetonic">@tswetonic</a> ,</p>
<p>For actions/setup-ruby, the required minimum version is 2.4. You can find it mentioned in action.yaml <a href="https://github.com/actions/setup-ruby/blob/458abf6134c3aa66f6fd0961639a643a2f4ea9a5/action.yml#L7" rel="nofollow noopener">here</a>.</p>
<p>You can use another action ‘ruby/setup-ruby@v1’  instead. Code as below, hope it helps.</p>
<pre><code>- uses: ruby/setup-ruby@v1
      with:
        ruby-version: 2.3.4
    - run: ruby -v
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/62de1ec767ead199013d8d16f8494d4cf0546ec4.png" data-download-href="/uploads/short-url/e6CGjxqYa2yX3YARDQ9O9agiPmQ.png?dl=1" title="rubysetup.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181102131-f119225f-801a-4617-964b-a6d60f000061.png" alt="rubysetup.png" data-base62-sha1="e6CGjxqYa2yX3YARDQ9O9agiPmQ" width="690" height="207" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/62de1ec767ead199013d8d16f8494d4cf0546ec4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">rubysetup.png</span><span class="informations">1090×327 16.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tswetonic** [comment]
<p>I tried that and it didn’t work. Will try again, thanks! I did have success with rvm too.</p>

**Comment by tswetonic** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> did you use ubuntu-latest as your runner?</p>

**Comment by tswetonic** [comment]
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a> this now works. Last time I tried it, it failed!</p>

**Comment by tswetonic** [comment]
<p>Not sure what changed. Maybe this was a one-off failure when I tried it.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/15eca850c200dc1861706073b4ab88ded84cd91c.png" data-download-href="/uploads/short-url/37X3UVyyFGM0OkptSKPcsBrazQU.png?dl=1" title="notfound.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/54121763/181102136-71385613-c1b0-4867-a1c6-94b8b9e66d11.png" alt="notfound.png" data-base62-sha1="37X3UVyyFGM0OkptSKPcsBrazQU" width="690" height="170" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/15eca850c200dc1861706073b4ab88ded84cd91c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">notfound.png</span><span class="informations">712×176 16.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by amein0** [comment]
<p>نظر شما رو برسی میکنم با تشکر</p>
